tests/keras/test_keras_model_export.py

```diff
@@ -98,6 +146,47 @@ def test_model_save_load(model, model_path, data, predicted):
         np.array(spark_udf_preds), predicted.reshape(len(spark_udf_preds)), decimal=4)
 
 
+@pytest.mark.large
+def test_custom_model_save_load(custom_model, custom_layer, data, custom_predicted, model_path):
+    x, _ = data
+    custom_objects = {'MyDense': custom_layer}
+    mlflow.keras.save_model(custom_model, model_path, custom_objects=custom_objects)
+
+    # Loading Keras model
+    model_loaded = mlflow.keras.load_model(model_path)
+    assert all(model_loaded.predict(x) == custom_predicted)
+    # pyfunc serve
+    scoring_response = pyfunc_serve_and_score_model(
+        model_uri=os.path.abspath(model_path),
+        data=pd.DataFrame(x),
+        content_type=pyfunc_scoring_server.CONTENT_TYPE_JSON_SPLIT_ORIENTED)
+    assert np.allclose(
+            pd.read_json(scoring_response.content, orient="records").values.astype(np.float32),
+            custom_predicted,
+            rtol=1e-5,
+            atol=1e-9)
+    # Loading pyfunc model
+    pyfunc_loaded = mlflow.pyfunc.load_pyfunc(model_path)
+    assert all(pyfunc_loaded.predict(x).values == custom_predicted)
+    # test spark udf
+    spark_udf_preds = score_model_as_udf(model_uri=os.path.abspath(model_path),
+                                         pandas_df=pd.DataFrame(x),
+                                         result_type="float")
+    np.testing.assert_array_almost_equal(
+        np.array(spark_udf_preds), custom_predicted.reshape(len(spark_udf_preds)), decimal=4)
+
+
+def test_custom_model_save_respects_user_custom_objects(custom_model, custom_layer, model_path):
+    class DifferentCustomLayer():
+        def __init__(self):
+            pass
+    incorrect_custom_objects = {'MyDense': DifferentCustomLayer()}
+    correct_custom_objects = {'MyDense': custom_layer}
+    mlflow.keras.save_model(custom_model, model_path, custom_objects=incorrect_custom_objects)
+    model_loaded = mlflow.keras.load_model(model_path, custom_objects=correct_custom_objects)
```

Comment:
Can we also verify that calling `load_model` with `incorrect_custom_objects` throws an exception?